### PR TITLE
Refactor/reorganization

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
1. Move compiled files to `dist/`
2. Use [prettier](https://prettier.io/) for formatting
3. Change compilation target to more recent NodeJS version

These changes are opiniated and not mandatory.

### CODE REVIEW

File: AssignCardToSender/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { OctokitAction } from '../lib/OctokitAction';
```
Comment (pavel-mikula-sonarsource) [inline AssignCardToSender/index.ts:1]: It is intentional that these files are not named `index.ts`, because that would be difficult to work with in dev UI
image

vs. `index.js`, `index.js`, `index.js` - you don't know what you're editing and where to go next

File: AssignCardToSender/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+class AssignCardToSender extends OctokitAction {
```
Comment (pavel-mikula-sonarsource) [inline AssignCardToSender/index.ts:3]: Related to git work: When renaming file,s you should stage them, so git can recognize rename for the commit history.

It will also enable to see what was the actual formatting change, if any.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
The `index.ts` was intentional. The rest of the changes looks good.

File: AssignCardToSender/index.ts (hunk 1 above)
Comment (ilia-kebets-sonarsource) [reply AssignCardToSender/index.ts:1]: Yeah, I thought about this afterwards, indeed I had a similar discussion a few years back. I wouldn't mind keeping the name of the action as the name of file, even if it means repetition.
I will roll this back.

File: AssignCardToSender/index.ts (hunk 2 above)
Comment (ilia-kebets-sonarsource) [reply AssignCardToSender/index.ts:3]: Yeah, I guess, I got a bit confused in my git commits with the various renamings and deletions, which led to git not recognizing it as a single renaming action.

Comment (ilia-kebets-sonarsource) [reply AssignCardToSender/index.ts]: It might also have confused git that I ran the `prettier` formatter in addition to moving the files.

**Comment by ilia-kebets-sonarsource** [comment]
@pavel-mikula-sonarsource I rolled back the `ActionName.ts => index.ts` renaming and pulled your latest changes.

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@
 /obj/
 
 *.user
+#node_modules
```
Comment (pavel-mikula-sonarsource) [inline .gitignore:6]: Is this change intentional? It doesn't seem so

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
This is too hard to review, and the commit history doesn't help much :/ Can you split it?

I'd suggest creating several PRs instead, with a linear history:
* One to add prittier dependency - just to have the `node_modules` there, dont' use it.
* Another to change the "Dist" directory. This should have all files marked as renamed, instead of delete&add.
* Another to change the file permissions in `node_modules/@types/node/*` files 
* Another to apply the actual prettier formatting changes.

File: .gitignore (hunk 3 above)
Comment (ilia-kebets-sonarsource) [reply .gitignore:6]: probably not, thanks!

**Comment by ilia-kebets-sonarsource** [comment]
I split this PR into 5: 
1. https://github.com/SonarSource/gh-action-lt-backlog/pull/18
2. https://github.com/SonarSource/gh-action-lt-backlog/pull/19
3. https://github.com/SonarSource/gh-action-lt-backlog/pull/20
4. https://github.com/SonarSource/gh-action-lt-backlog/pull/21
5. https://github.com/SonarSource/gh-action-lt-backlog/pull/22
